test/components/views/dialogs/CreateRoomDialog-test.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { fireEvent, render, screen, within } from "@testing-library/react";
+import { Preset, Visibility } from "matrix-js-sdk/src/matrix";
+
+import CreateRoomDialog from "../../../../src/components/views/dialogs/CreateRoomDialog";
+import { flushPromises, getMockClientWithEventEmitter, mockClientMethodsUser } from "../../../test-utils";
+
+// mock so we can snapshot test fields
+jest.mock("matrix-js-sdk/src/randomstring", () => ({
+    randomString: () => "not-a-random-string",
+}));
+
+describe("<CreateRoomDialog />", () => {
+    const userId = "@alice:server.org";
+    const mockClient = getMockClientWithEventEmitter({
+        ...mockClientMethodsUser(userId),
+        getDomain: jest.fn().mockReturnValue("server.org"),
+        getClientWellKnown: jest.fn(),
+        doesServerForceEncryptionForPreset: jest.fn(),
+        // make every alias available
+        getRoomIdForAlias: jest.fn().mockRejectedValue({ errcode: "M_NOT_FOUND" }),
+    });
+
+    const getE2eeEnableToggle = () => screen.getByTestId("enableE2ee");
```

Comment:
Could we use `getByLabelText` here instead?

Reply:
Done! Thanks for the early morning review :-)